### ISSUE TITLE
Advertise appropriate server version to MySQL clients for non-TLS routing mode environment. 

### ISSUE THREAD

**Issue by oshati** [gravitational/teleport#13734]
What would you like Teleport to do?
Advertise appropriate server version to MySQL clients for non-TLS routing mode environment. 

What problem does this solve?
This is primarily causing issues for applications that use DataGrip GUI clients to connect to a mysql 8.0+ database.

If a workaround exists, please include it.
Setup works without TLS routing mode enabled.

gz#5603

**Comment by r0mant** [comment]
Since MySQL protocol is server initiated (when client connects, the server sends the first handshake message containing server version), Teleport cannot report the proper version without TLS routing.

GUI and other clients that need to receive the correct version must connect through the TLS routing local proxy started with `tsh proxy db` or using [Teleport Connect](https://get.gravitational.com/Teleport%20Connect-10.0.0.dmg).

There is an issue with TLS routing, being dependent on mutual TLS, not working in certain deployment scenarios, for example when Teleport Proxy is behind an ALB. We're working on addressing this by adding ability to `tsh proxy db` to tunnel mTLS connections over websockets which should allow to use TLS routing even in setups where terminating load balancers are used.

cc @smallinsky

**Comment by keenan-v1** [comment]
@r0mant - Curious, there's a way to tell the database service what version it should report via configuration. This is not working as of 9.3.7. Is this something new in v10 or is it something that can be made to work in a future update? That would also solve the problem since I can easily query our infrastructure for the current running engine version. I actually do that already as a failed attempt to fix this issue.

**Comment by eitah** [comment]
@keenan-v1 I understand the version configuration only works once a user is already on TLS multiplexing. For most of us that feature is off so the version configuration is ignored. (Reference: Slack thread with teleport https://goteleport.slack.com/archives/CEZH6UL64/p1655134785659229)

As a workaround at $CURRENT_JOB, when users are unable to connect to MySQL via Datagrip today, we patch the compliled teleport binary to transmit the below string instead of the current `8.0.0-Teleport`. This remediates connectivity for MySQL 5.7+ databases but breaks connectivity for MySQL 5.6 (which is EOL so we don't feel too bad). Maybe it will help you workaround your use case?

```Server version: 8.0.9-Teleport (Google)```

**Comment by smallinsky** [comment]
@eitah 
> I understand the version configuration only works once a user is already on TLS multiplexing

Right now there is a `tsh proxy db --tunnel mysql02` command that allows forcing TLS Routing connection even if TLS `multiplex` mode was not set (currently we are working on an improvement to support `tsh proxy db` when Teleport proxy is behind ALB) 

In the case of `multiplex` mode is disabled the `tsh db connect` and `tsh db config` will result a connection to a separate MySQL Teleport proxy listener where the MySQL Server version is hardcoded (`8.0.0-Teleport`):
```sh
tsh db connect --db-user=marek mysql02
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 10003
Server version: 8.0.0-Teleport Source distribution
```

But if  Teleport proxy is not behind ALB a user can establish a database  `tunnel`  TLS Routing connection to the Teleport Proxy:  
```sh
tsh proxy db --port 9999 --tunnel mysql02
Started authenticated tunnel for the MySQL database "mysql02" in cluster "main" on 127.0.0.1:9999.

Use the following command to connect to the database:
  $  /usr/local/opt/mysql-client/bin/mysql --user marek --port 9999 --host localhost --protocol TCP
  ```
  In the result a connection to the teleport database local proxy made from GUI client or CLI should report the correct MySQL Server version: 
  ```sh
  /usr/local/opt/mysql-client/bin/mysql --user marek --port 9999 --host localhost --protocol TCP
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 10004
Server version: 8.0.26 Source distribution
```

**Comment by keenan-v1** [comment]
My proxy is behind an ALB, so it looks like tunnels are not on the table for me. I know I can fork and "fix" the issue, but I don't want to maintain a fork of Teleport for such a small thing. If it cannot read the server version from the connection itself, then at least honor the configuration option so I can work around this.

**Comment by smallinsky** [comment]
@keenan-v1 
We are planning to add support for Teleport proxy behind ALB  in https://github.com/gravitational/teleport/issues/14369 which will solve the MySQL version propagation. 

The problem with the proxy MySQL version configuration parameter is the fact that this will be settable per teleport proxy so in case of many different MySQL instances connected to the teleport cluster we will advertise only one MySQLSQL server version for all database connections so this is only a partial solution.

**Comment by keenan-v1** [comment]
@smallinsky
Thank you for the update; I look forward to the support coming! Let me go into how I have our Teleport set up so I can better understand the `server_version` configuration option.

I consider our usage of Teleport to include four pieces:
* Auth Service
* Proxy Service
* Host Agent
* Database Service

The way I have things configured, the auth service is strictly internal and only accessible by the proxy service. The proxy service is what is public-facing and behind the ALB. I then have a "database service" container, which is just a teleport agent running in a container with access to the database it provides connectivity. The host agents only provide ssh access and run on the host instances that run the containers.

Given this configuration, I expected the `db_service.mysql.server_version` configuration option to be the version it advertises. This works in my case since the database service only speaks to a single database. It also works since we use RDS, and I can query the AWS API for the engine version for a given database.

We are using v9.3.7 at the moment.

**Comment by smallinsky** [comment]
The fix for mysql version propagation was provided by  on https://github.com/gravitational/teleport/pull/15669 and is available from Teleport 10.2.4  release.
